internal/upload/koji/koji.go

```diff
@@ -284,13 +283,32 @@ func (k *Koji) CGImport(build ImageBuild, buildRoots []BuildRoot, images []Image
 		return nil, err
 	}
 
-	var result CGImportResult
-	err = k.xmlrpc.Call("CGImport", []interface{}{string(metadata), directory, token}, &result)
-	if err != nil {
-		return nil, err
+	const retryCount = 10
+	const retryDelay = time.Second
+
+	for attempt := 0; attempt < retryCount; attempt += 1 {
+		var result CGImportResult
+		err = k.xmlrpc.Call("CGImport", []interface{}{string(metadata), directory, token}, &result)
+
+		if err != nil {
+			// Retry when the error mentions a corrupted upload. It's usually
+			// just because of NFS inconsistency when the kojihub has multiple
+			// replicas.
+			if strings.Contains(err.Error(), "Corrupted upload") {
+				time.Sleep(retryDelay)
+				continue
```

Comment:
I'm sure this works, but I'm wondering it it would be worth to use exponential backoff time instead of a constant for each retry? 🤔 It could result in less retries...

IOW, it would be interesting to see how many retries (seconds) it takes on average to make the import successful.

Reply:
I added a logging statement, is that enough? Exponential back-off seems nice, but tbh, I feel lazy to implement it. What about merging this, seeing how this behaves in the real world and then adjusting it if needed?